cmd/login.go

```diff
@@ -3,27 +3,77 @@ package cmd
 import (
 	"fmt"
 	"os"
+	"os/user"
+	"time"
 
 	"github.com/spf13/afero"
 	"github.com/spf13/cobra"
 	"github.com/supabase/cli/internal/login"
-	"github.com/supabase/cli/internal/utils"
+	"golang.org/x/term"
 )
 
+func generateTokenName() (string, error) {
+	user, err := user.Current()
+	if err != nil {
+		return "", fmt.Errorf("cannot retrieve username: %w", err)
+	}
+
+	hostname, err := os.Hostname()
+	if err != nil {
+		return "", fmt.Errorf("cannot retrieve hostname: %w", err)
+	}
+
+	return fmt.Sprintf("cli_%s@%s_%d", user.Username, hostname, time.Now().Unix()), nil
+}
+
 var (
 	loginCmd = &cobra.Command{
 		GroupID: groupLocalDev,
 		Use:     "login",
 		Short:   "Authenticate using an access token",
 		RunE: func(cmd *cobra.Command, args []string) error {
-			return login.Run(os.Stdin, afero.NewOsFs())
-		},
-		PostRun: func(cmd *cobra.Command, args []string) {
-			fmt.Println("Finished " + utils.Aqua("supabase login") + ".")
+			params := login.RunParams{
+				Fsys: afero.NewOsFs(),
+			}
+
+			if cmd.Flags().Changed("token") {
+				token, err := cmd.Flags().GetString("token")
+				if err != nil {
+					return fmt.Errorf("cannot parse 'token' flag: %w", err)
+				}
+				params.Token = token
+			}
+
+			if cmd.Flags().Changed("name") {
+				name, err := cmd.Flags().GetString("name")
+				if err != nil {
+					return fmt.Errorf("cannot parse 'name' flag: %w", err)
+				}
+				params.Name = name
+			} else {
+				name, err := generateTokenName()
+				if err != nil {
+					params.Name = fmt.Sprintf("cli_%d", time.Now().Unix())
+				} else {
+					params.Name = name
+				}
+			}
+
+			if cmd.Flags().Changed("no-browser") {
+				params.OpenBrowser = false
+			} else {
+				params.OpenBrowser = term.IsTerminal(int(os.Stdin.Fd()))
+			}
+
+			return login.Run(os.Stdin, params)
 		},
 	}
 )
 
 func init() {
+	loginFlags := loginCmd.Flags()
+	loginFlags.String("token", "", "Use provided token instead of automatic login flow")
```

Comment:
Passing in the token via cli flag seems insecure because it could be saved in shell history. Shall we only support `SUPABASE_ACCESS_TOKEN` env var?

Reply:
I used `--auth-token` flags in the past extensively and not a single person reported this as an issue tbh